### ISSUE TITLE
If a human can't perform the task with ease, be surprised if a machine can. 

### ISSUE THREAD

**Issue by paulbrodersen** [Benjamin-Lee/deep-rules#26]
**Have you checked the [list of proposed rules](https://github.com/Benjamin-Lee/deep-rules/issues?q=is%3Aissue+is%3Aopen+label%3Arule) to see if the rule has already been proposed?**

- [x] Yes

**Feel free to elaborate, rant, and/or ramble.**

Alternative heading: Garbage in -- garbage out. 

Some people confuse machine learning -- and deep learning is no exception -- with elaborate statistical methods, and belief that neural networks have magical, super-human powers that can detect patterns nobody else can. As a consequence they belief that they can take their data in any form, throw the trending machine learning method of the day at it (which will automagically find a suitable representation of the data and then learn the task), and then profit.  

Life rarely works out that way. Humans are actually incredibly good at detecting patterns. More often than not, if a human can't see "it", a machine won't either (c.f. the rule on the importance on baseline performances). If the machine performs better than chance, be suspicious. It is probably overfitting the training data (c.f. the rule on freezing a test set). For success, it is generally tantamount to find a representation of the data based on which the human could perform the task, and then -- and only then -- try to teach a machine to do it. 

**Any citations for the rule?** (peer-reviewed literature preferred but not required)
- [DOI](doi.org/DOI_goes_here)

**Comment by hugoaerts** [comment]
In general I agree with this. If humans cant do it, likely the information is not in the data. However, there are also examples of machines that do better job compared to humans. For example, look at the lung screening Kaggle challenge. Benign or malignant lung nodules look very similar to humans, but machines can discriminate them. But I like the rule, but we should note that machine simply can do some things better, but we need appropriate validation/testing of the networks to reduce false positive results.

**Comment by paulbrodersen** [comment]
Yeah, that exact example motivated the qualifier: 

> More often than not, [...]

However, the fact that we had the exact same example in mind is maybe telling that there aren't that many: a deafening silence, if you will, only broken by a few whispers (so far). 
Hence my advice for caution: 

> If a human can't perform the task with ease, be surprised if a machine can.

**Comment by agitter** [comment]
> More often than not, if a human can't see "it", a machine won't either

What type of data are you thinking of here?

I agree that we should counter the belief that DL will automagically work, but there are many types of high-dimensional biological data where humans cannot easily find patterns and ML/DL can.  Successful applications of autoencoders come to mind.

**Comment by cgreene** [comment]
DeepSEA, Basset, etc all seem to find patterns that humans wouldn't notice by looking at a lot of sequences.

**Comment by agitter** [comment]
I find the transcriptomics applications to be more convincing, even if Casey [has shown](https://doi.org/10.1101/385534) their dependence on parameters.

In biochemistry / drug discovery, [this autoencoder](http://doi.org/10.1021/acscentsci.7b00572) is another example of finding some structure in a domain that is unapproachable via manual inspection.  It may not be a perfect latent representation, but there is learned structure there.

> DeepSEA, Basset, etc all seem to find patterns that humans wouldn't notice by looking at a lot of sequences.

Yes, regulatory genomics is another example where NNs find meaningful patterns that are not detectable by humans.  That doesn't mean that we can blindly trust neural networks or shouldn't evaluate them with skepticism.  However, The ENCODE-DREAM in vivo Transcription Factor Binding Site Prediction Challenge comes to mind as one specific example where careful analysis revealed a lot about context dependence of TF sequence preferences and other real biology.

**Comment by evancofer** [comment]
@jmschrei I would disagree, since the application/use of these models is not just limited to identification of regulatory patterns. For instance, Basenji/ExPecto modeled gene expression from regulatory signals, a task for which there is little reason to expect good human performance. Most of those models seem to outperform conservation-based methods for identifying the effects of regulatory variants (both SNPs and indels), which humans are generally bad at. It may be possible to identify regulatory effects of common variants through assocation-based studies, but these DL methods are the best approach to analyzing rare or de novo variants.

**Comment by jmschrei** [comment]
My comment wasn't to say that you should be surprised if deep networks outperform humans on tasks that humans do well at, but you should be surprised when deep networks perform well on tasks humans can't do. If humans were totally unable to model gene expression using sequence alone and yet a deep model came out that could do it, I'd be very skeptical. However, many of these models validate their predictions by showing that when making known mutations to TF binding sites you get an expected change in gene expression. This is because humans are capable of identifying associations, but certainly not as quickly as one could train a deep model.

**Comment by evancofer** [comment]
@jmschrei I agree with that sentiment about rigorous statistical evaluation. I would also go a step further and say that claims of novel biological mechanisms and phenomena should require experimental demonstration, but this is more of a general ML/data science rule than a DL-specific one.
